language_detector/main.py

```diff
@@ -1,9 +1,53 @@
 """This is the entry point of the program."""
 
-from .languages import LANGUAGES
+try:
+    from .languages import LANGUAGES
+except:
+    from languages import LANGUAGES
 
 
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    spanish = []
+    english = []
+    german = []
+    #Grant easy access to the dictionaries
+    for lang in languages:
+        if lang["name"] == "Spanish":
+            spanish = lang["common_words"]
+        elif lang["name"] == "English":
+            english = lang["common_words"]
+        elif lang["name"] == "German":
+            german = lang["common_words"]
+            
+    #Create a variable with number of occurrences per language
+    english_occurrences = 0
+    german_occurrences = 0
+    spanish_occurrences = 0
+    
+    #Tokenize the text input
+    tokenized_text = text.split(" ")
+
+    
+    #count occurrences
+    #So we could iterate through that list
+    for word in tokenized_text:
+        #If the word is contained in the english common words list
+        if word in english:
+            #Increase the english occurrences
+            english_occurrences += 1
+            
+        elif word in spanish:
+            #increase the spanish occurences
+            spanish_occurrences += 1
+        elif word in german:
+            #increase the german occurences
+            german_occurrences += 1
```

Comment:
Jon makes a good point. You are assuming that the common_words lists for each language are mutually exclusive. Also the languages here are once again hard coded. If you use the count_dict dictionary idea described above, you could iterate through all the languages and update the values in that dictionary to keep a score tally. Something like:. 

```python 
#Iterate through all language dictionaries contained in the languages list:
for langdict in languages:
         langname = langdict['name']
         wordlist = langdict['common_words']
        #Now go thorugh all the words in the block of text and check if each word is in the wordlist
         for word in text.lower().split():
             if word in wordlist:
                 #If the word is in the wordlist, update the count_dict score for given language by 1
                 count_dict[langname] += 1
```